File: typescript/src/array/merge-intervals.ts
function merge(intervals: number[][]): number[][] {
  if (intervals.length < 2) return intervals;
  intervals = intervals.sort((a, b) => a[0] - b[0]);
  let result: number[][] = [];
  let workingInterval = intervals[0];

  for (let i = 1; i < intervals.length; i++) {
    if (workingInterval[1] >= intervals[i][0]) {
      workingInterval = [
        workingInterval[0],
        Math.max(workingInterval[1], intervals[i][1])
      ];
    } else {
      result.push(workingInterval);
      workingInterval = intervals[i];
    }
  }
  result.push(workingInterval);
  return result;
};

File: cs/LeetCode/LeetCode/_Array/_MergeIntervals.cs


namespace LeetCode._Array
{
    internal class _MergeIntervals
    {
        public int[][] Merge(int[][] intervals)
        {
            if (intervals.Length < 2)
                return intervals;
            Array.Sort(intervals, (a, b) => a[0] - b[0]);
            List<int[]> result = new List<int[]>();
            int[] workingInterval = intervals[0];

            for (int i = 1; i < intervals.Length; i++)
            {
                if (workingInterval[1] >= intervals[i][0])
                {
                    workingInterval[0] = Math.Min(workingInterval[0], intervals[i][0]);
                    workingInterval[1] = Math.Max(workingInterval[1], intervals[i][1]);
                }
                else
                {
                    result.Add(workingInterval);
                    workingInterval = intervals[i];
                }
            }
            result.Add(workingInterval);
            return result.ToArray();
        }
    }
}
